decidim-core/spec/commands/decidim/authorize_user_spec.rb

```diff
@@ -37,5 +34,39 @@ module Decidim
         expect(user.authorizations.first.metadata["document_number"]).to eq("12345678X")
       end
     end
+
+    describe "uniqueness" do
+      let(:user) { create(:user) }
+      let(:unique_id) { "foo" }
+
+      before do
+        Decidim.authorization_handlers.push(DummyAuthorizationHandler)
+      end
+
+      after do
+        Decidim.authorization_handlers.delete(DummyAuthorizationHandler)
+      end
+
+      context "when there's no other authorizations" do
+        it "is valid if there's no authorization with the same id" do
+          expect { subject.call }.to change { user.authorizations.count }.by(1)
+        end
+      end
+
+      context "when there's other authorizations" do
+        let!(:other_user) { create(:user, organization: user.organization)}
+
+        before do
+          create(:authorization,
+                 user: other_user,
+                 unique_id: document_number,
+                 name: handler.handler_name)
+        end
+
+        it "is invalid if there's another authorization with the same id" do
+          expect { subject.call }.to change { user.authorizations.count }.by(0)
```

Comment:
Well yeah but I didn't know how to expect two things at once xD. What I actually wanted to make sure is I don't create the authorization.